frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -798,22 +836,366 @@ class SegmentsView extends React.Component<Props, State> {
+    });
+  };
+
+  handleRemoveMeshes = (groupId: number) => {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
+    this.handlePerSegment(groupId, (segment) => {
+      if (
+        Store.getState().localSegmentationData[visibleSegmentationLayer.name].isosurfaces[
+          segment.id
+        ] != null
+      ) {
+        Store.dispatch(removeIsosurfaceAction(visibleSegmentationLayer.name, segment.id));
+      }
+    });
+  };
+
+  handleChangeMeshVisibilityInGroup = (layerName: string, groupId: number, isVisible: boolean) => {
+    this.handlePerSegment(groupId, (segment) => {
+      if (Store.getState().localSegmentationData[layerName].isosurfaces[segment.id] != null) {
+        Store.dispatch(updateIsosurfaceVisibilityAction(layerName, segment.id, isVisible));
+      }
+    });
+
+    const newSegmentsInGroupVisible = {
+      ...this.state.areSegmentsInGroupVisible,
+      [groupId]: isVisible,
+    };
+    this.setState({
+      areSegmentsInGroupVisible: newSegmentsInGroupVisible,
+    });
+  };
+
+  handleLoadMeshesAdHoc = (groupId: number) => {
+    this.handlePerSegment(groupId, (segment) => {
+      if (segment.somePosition == null) return;
+      this.props.loadAdHocMesh(segment.id, segment.somePosition);
+    });
+
+    const newSegmentsInGroupVisible = {
+      ...this.state.areSegmentsInGroupVisible,
+      [groupId]: true,
+    };
+    this.setState({
+      areSegmentsInGroupVisible: newSegmentsInGroupVisible,
+    });
+  };
+
+  getSegmentsWithMissingLocation = (groupId: number): number[] => {
+    const segmentGroup = this.getSegmentsOfGroup(groupId);
+    if (segmentGroup == null) return [];
+    let segmentIdsWithoutPosition: number[] = [];
+    segmentGroup.forEach((segment) => {
+      if (segment.somePosition == null) segmentIdsWithoutPosition.push(segment.id);
+    });
+    return segmentIdsWithoutPosition.sort();
+  };
+
+  handlePerSegment(groupId: number, callback: (s: Segment) => void) {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
+    const segmentGroup = this.getSegmentsOfGroup(groupId);
+    if (segmentGroup == null) return;
+    segmentGroup.forEach(callback);
+  }
+
+  handleLoadMeshesFromFile = (groupId: number) => {
+    this.handlePerSegment(groupId, (segment: Segment) => {
+      if (segment.somePosition == null || this.props.currentMeshFile == null) return;
+      this.props.loadPrecomputedMesh(
+        segment.id,
+        segment.somePosition,
+        this.props.currentMeshFile.meshFileName,
+      );
+    });
+
+    const newSegmentsInGroupVisible = {
+      ...this.state.areSegmentsInGroupVisible,
+      [groupId]: true,
+    };
+    this.setState({
+      areSegmentsInGroupVisible: newSegmentsInGroupVisible,
+    });
+  };
+
+  downloadAllMeshesForGroup = (groupId: number) => {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
```

Comment:
Combine both return conditions into one conditional statement to keep it short

Reply:
I had the same thought as above so I'm looking forward to an answer!